hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAttributesValidator.java

```diff
@@ -73,6 +75,14 @@ public void validateTokenName(@Nullable final String name, @NonNull final Tokens
         tokenStringCheck(name, tokensConfig.maxTokenNameUtf8Bytes(), MISSING_TOKEN_NAME, TOKEN_NAME_TOO_LONG);
     }
 
+    /**
+     * Validates the token metadata, if it is exists and is not empty or not too long.
+     * @param metadata the token metadata to validate
+     */
+    public void validateTokenMetadata(@Nullable final String metadata, @NonNull final TokensConfig tokensConfig) {
```

Comment:
Related to a previous comment, is it ok in _all_ cases to convert the metadata bytes to a String?

Also for this class, would like to see unit tests for this method. Possible cases to exercise, with the following as metadata:
* a random string of bytes
* an empty byte array/Bytes.EMPTY
* a null value
* UTF-8 text, possibly including emojis :)

Reply:
Good call. I've added all the test cases now including exercising byte arrays with null, randoms and empty values.

Reply:
Looks great, thanks!